x/emissions/module/rewards_v2.go

```diff
@@ -0,0 +1,430 @@
+package module
+
+import (
+	"math"
+
+	errors "cosmossdk.io/errors"
+	"github.com/allora-network/allora-chain/x/emissions/types"
+)
+
+// SmoothAbsoluteX calculates the smooth absolute function.
```

Comment:
is this phi? it looks like phi over an array

Reply:
yeah, deleted it to use phi